lib/resources/node.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	log "github.com/sirupsen/logrus"
+
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// FindNodeAddress returns node address of the specified type. Type can be one of
+// CalicoNodeIP, InternalIP or ExternalIP
+func FindNodeAddress(node *apiv3.Node, ipType string) (*cnet.IP, *cnet.IPNet) {
```

Comment:
Rather than pass a string here, we should probably use the apiv3 type instead.

Reply:
Oh, right. I see that they are just defined as strings: https://github.com/projectcalico/libcalico-go/blob/master/lib/apis/v3/node.go#L26-L28

Normally we define a type for enums like this so that we can pass around typed values, like we do here: https://github.com/projectcalico/libcalico-go/blob/master/lib/apis/v3/policy.go#L21-L27

I suppose in this case the Kubernetes API defines the equivalent field as a string, so maybe this is OK.